crates/biome_service/src/configuration.rs

```diff
@@ -258,6 +259,36 @@ fn load_config(
     }
 }
 
+pub fn load_editorconfig(
+    file_system: &DynRef<'_, dyn FileSystem>,
+    workspace_root: PathBuf,
+) -> Result<
+    (
+        Option<PartialConfiguration>,
+        Vec<EditorConfigValidationError>,
+    ),
+    WorkspaceError,
+> {
+    if let Some(auto_search_result) =
+        match file_system.auto_search(&workspace_root, [".editorconfig"].as_slice(), false) {
+            Ok(result) => result,
+            Err(error) => return Err(WorkspaceError::from(error)),
+        }
+    {
+        let AutoSearchResult {
+            content,
+            file_path: _path,
+        } = auto_search_result;
+
+        let edconfig = biome_configuration::editorconfig::parse_str(&content)
+            .map_err(ConfigurationDiagnostic::EditorConfigError)?;
```

Comment:
Technically, `.editorconfig` **isn't** part of our configuration. I wouldn't want to mix them. I see two options to resolve this comment:
- Make `ConfigurationDiagnostic` a new enum like this and rename `EditorConfigError` to `EditorConfigDiagnostic`. A diagnostic isn't an error:
	```rs
	enum ConfigurationDiagnostic { 
		BiomeConfiguration(BiomeDiagnostic)
		EditorConfig(EditorConfigDiagnostic)
	}
	
	enum BiomeDiagnostic {} // the current ConfigurationDiagnostic
	enum EditorConfigDiagnostic {}
	```
- Make `EditoConfigDiagnostic` part of the `WorkspaceError` enum

I like more the first option but it requires more changes. I don't have strong options though, even the second option is fine.

Reply:
I'll do the second option for now, and follow this up with a refactor for the first option.